src/plugins/chart_expressions/expression_xy/common/expression_functions/layered_xy_vis_fn.ts

```diff
@@ -22,7 +22,12 @@ import { appendLayerIds, getDataLayers } from '../helpers';
 export const layeredXyVisFn: LayeredXyVisFn['fn'] = async (data, args, handlers) => {
   const layers = appendLayerIds(args.layers ?? [], 'layers');
 
-  logDatatables(layers, handlers);
+  // for visialize we should log one datable for all layers
+  if (handlers.getExecutionContext()?.name === 'visualize') {
```

Comment:
To avoid checking for visualize here, let's instead check whether the tables are the same:
```
const sameTable = layers.every(l => l.table === layers[0].table)
```

Reply:
Unless in Lens there can be a situation that there will be an identical table in two layers?

Reply:
@VladLasitsa This can't happen in Lens - layers always have their separate data fetching

Reply:
And even if it would happen IMHO it makes sense to report it as a single table in the inspector if it is virtually the same object.

Reply:
Done.

Reply:
@flash1293 I have checked this more deeply and found out that we should use my approach because for lens we should log datatable using layer id as key, but when we log only one datatable we use "default" as key. It leads to some problems with getting correct datatable from inspector which we use after as active data in many places.